### PR TITLE
Make the nav2_common Node action use the launch_ros Node as super class

### LINKED ISSUE #1466: Question: Why do we have a custom Node launch class?
In `nav2_common` we have a separate `Node` that may make it hard to transition to lifecycle nodes. Why is this here? I'm not seeing alot of documentation about it. I assume it has to do with the remappings, but this should be explicit somewhere and what a path might be back to using the base-Node launch tool.

### LINKED ISSUE #1474: nav2_common.launch.Node passing empty arguments to process, causing process to crash
## Bug report

**Required Info:**

- Operating System:
  - Ubuntu 18.04
- Version or commit hash:
  - nav2 and ros2 latest as of about 9am 16 Jan 2020
- DDS implementation:
  - 

#### Steps to reproduce issue
1. `ros2 launch nav2_bringup nav2_tb3_simulation_launch.py`

#### Expected behavior
nodes launch and run

#### Actual behavior
Many nodes crash with an error like below
```
[planner_server-8] terminate called after throwing an instance of 'rclcpp::exceptions::UnknownROSArgsError'
[planner_server-8]   what():  found unknown ROS arguments: '', '', '', '', '', '', '', ''
```
#### Additional information

nav2_common.launch.Node is adding '' arguments here
https://github.com/ros-planning/navigation2/blob/4cf60e9d342adf71bef5c0a60faa7c686c0f86fa/nav2_common/nav2_common/launch/node.py#L374

This doesn't seem to affect all machines. I'm not sure what the difference is.

### PR DESCRIPTION
**Pull request by crdelsey**
## Basic Info

| Info | Please fill out this column |
| ------ | ----------- |
| Ticket(s) this addresses   | fixes #1474, fixes #1466 |
| Primary OS tested on | Ubuntu |
| Robotic platform tested on | TB3 in Gazebo|

---

## Description of contribution in a few bullet points
* Rather than reimplement the launch_ros Node action, our custom class now derives from it.

### CODE REVIEW

**Comment by crdelsey** [comment]
> Killing 3 tickets and deleting code?

Well, I didn't actually get #1475. Some of the tests are still failing, but for unrelated reasons I believe. I tried to make up for it by deleting some more code though. ;-)

File: nav2_common/nav2_common/launch/node.py [hunk 1]
```diff
@@ -131,255 +88,10 @@ def __init__(
...
-                elif nested_params:
-                    param_dict.update({
-                        name: get_nested_dictionary_from_nested_key_value_pairs(nested_params)
-                    })
-                else:
-                    raise RuntimeError('either a value attribute or nested params are needed')
-            return param_dict
-
-        normalized_params = []
-        params_without_from = []
```
Comment (crdelsey) [inline nav2_common/nav2_common/launch/node.py]: @orduno I wasn't sure what you were doing here with the `params_without_from` and associated code. Can you explain what this was doing? It seems to work without, but I'm not sure if I'm missing a corner case.

**Comment by orduno** [comment]
@crdelsey do the changes work if remappings are provided (as we are currently always doing in the `nav2_bringup` launch files) but `use_remappings` is false?

I previously tried this option (deriving from `launch_ros::node`) but found some problems getting it to work for this case; as the space for remappings is [reserved on the constructor](https://github.com/ros2/launch_ros/blob/c5e2aa2bf0eb60d61c66dbe0a6da365230b3387d/launch_ros/launch_ros/actions/node.py#L155https://github.com/ros2/launch_ros/blob/c5e2aa2bf0eb60d61c66dbe0a6da365230b3387d/launch_ros/launch_ros/actions/node.py#L155).

So if we have, let's say three remappings, the [cmd passed to the base class](https://github.com/ros2/launch_ros/blob/c5e2aa2bf0eb60d61c66dbe0a6da365230b3387d/launch_ros/launch_ros/actions/node.py#L159)  already contains placeholders for these. During execution, these placeholders/substitutions get replaced with their actual value (empty if the flag is false) resulting in a launch command like `my_exec --ros_args -r := -r := -r :=` (maybe without the `:=`).

**Comment by crdelsey** [comment]
@orduno Thanks. I think I understand the problem now.

The current code on the PR does seem to work for both the use_remappings true and false cases. It's in fact the code on master that produces the error case you describe (or something like it).

That said, I don't think this PR is quite right yet due to the reserved parameters problem you mentioned. I'll see if I can find an alternative. I think the solution is to always use remappings, but use substitutions in the remappings to change the to and from values as needed. This could remove the need for a custom Node altogether.

**Comment by crdelsey** [comment]
This PR is ready for review now. In the latest version, I got rid of our custom Node action altogether and I also got rid of the use_remappings flag. Instead, we always provide the remapping that removes the leading slash from /tf and /tf_static.

The result is that if any of these nodes are not put in a namspace, tf and tf_static stay in the root namespace. If we push the nodes into the robot1 or robot2 namespace, the tf and tf_static topics end up in that namespace as well.

The one use case that isn't supported is if someone wants to push navigation into a namespace, but still use the root /tf tree. It's not a usecase we use, so I'm not sure if we need to support that.

File: nav2_bringup/bringup/launch/nav2_localization_launch.py [hunk 2]
```diff
@@ -35,17 +34,14 @@ def generate_launch_description():
...
     # Map fully qualified names to relative ones so the node's namespace can be prepended.
     # In case of the transforms (tf), currently, there doesn't seem to be a better alternative
     # https://github.com/ros/geometry2/issues/32
     # https://github.com/ros/robot_state_publisher/pull/30
     # TODO(orduno) Substitute with `PushNodeRemapping`
     #              https://github.com/ros2/launch_ros/issues/56
-    remappings = [((namespace, '/tf'), '/tf'),
-                  ((namespace, '/tf_static'), '/tf_static'),
-                  ('/tf', 'tf'),
+    remappings = [('/tf', 'tf'),
```
Comment (orduno) [inline nav2_bringup/bringup/launch/nav2_localization_launch.py]: Great! Not sure why I didn't do this simplification before. Sadly CI is broken to verify, but this should work. Remapping `/tf` to `tf` means that a node subscribing to `/tf` will effectively subscribe to `tf` (or `node_namespace/tf` and publishing to `/tf` will publish to `tf` (or `node_namespace/tf`).

**Pull request merged**